File: android/app/src/main/java/app/formstr/calendar/NotificationSchedulerPlugin.java
package app.formstr.calendar;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/** Small foreground bridge: all scheduling decisions stay in NotificationWorker. */
@CapacitorPlugin(name = "NotificationScheduler")
public class NotificationSchedulerPlugin extends Plugin {

    @PluginMethod
    public void reconcile(PluginCall call) {
        NotificationWorker.enqueueImmediate(getContext());
        call.resolve();
    }

    @PluginMethod
    public void clear(PluginCall call) {
        NotificationWorker.clearScheduledNotifications(getContext());
        call.resolve();
    }

    @PluginMethod
    public void cancelEvent(PluginCall call) {
        String eventId = call.getString("eventId");
        if (eventId == null || eventId.isEmpty()) {
            call.reject("eventId is required");
            return;
        }

        NotificationWorker.cancelEventNotifications(getContext(), eventId);
        call.resolve();
    }
}

File: ios/App/App/NotificationSchedulerPlugin.swift
import Capacitor

@objc(NotificationSchedulerPlugin)
public class NotificationSchedulerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "NotificationSchedulerPlugin"
    public let jsName = "NotificationScheduler"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "reconcile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "clear", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "cancelEvent", returnType: CAPPluginReturnPromise),
    ]

    @objc func reconcile(_ call: CAPPluginCall) {
        IOSNotificationScheduler.shared.reconcile {
            call.resolve()
        }
    }

    @objc func clear(_ call: CAPPluginCall) {
        IOSNotificationScheduler.shared.clear {
            call.resolve()
        }
    }

    @objc func cancelEvent(_ call: CAPPluginCall) {
        guard let eventId = call.getString("eventId"), !eventId.isEmpty else {
            call.reject("eventId is required")
            return
        }

        IOSNotificationScheduler.shared.cancel(eventId: eventId) {
            call.resolve()
        }
    }
}
